src/Cassandra/BatchStatement.cs

```diff
@@ -113,6 +113,11 @@ public BatchStatement SetRoutingValues(params object[] keys)
         /// <returns>The Batch statement</returns>
         public BatchStatement Add(Statement statement)
         {
+            if (_queries.Count >= ushort.MaxValue)
+            {
+                //see BatchMessage.codec field in BatchMessage.java in server code, and BatchRequest.GetFrame in this driver
+                throw new ArgumentOutOfRangeException(string.Format("There can be only {0} child statement in a batch statement accordung to the cassandra frame protocoll", ushort.MaxValue));
```

Comment:
Minor typos:
`"There can be only {0} child statement in a batch statement according to the Cassandra native protocol"`

Also, the [batch length is a `short`](https://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=doc/native_protocol_v3.spec) so it should be `short.MaxValue`